tests/backup/common.go

```diff
@@ -0,0 +1,211 @@
+				logrus.Warnf("ERR is %s", err)
+				if err != nil {
+					logrus.Errorf("An Error Occured while validating the pod %v", err)
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+func deleteRuleForBackup(orgID string, name string, uid string) bool {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]", err))
+	if err != nil {
+		logrus.Errorf("Failed to get context with error:%v", err)
+		return false
+	}
+	RuleDeleteReq := &api.RuleDeleteRequest{
+		Name:  name,
+		OrgId: orgID,
+		Uid:   uid,
+	}
+	_, err = Inst().Backup.DeleteRule(ctx, RuleDeleteReq)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create rules Error: [%v]", err))
+	if err != nil {
+		logrus.Errorf("Rules failed to get deleted with error: %v", err)
+		return false
+	}
+	return true
+}
+
+func contains(app_list []string, app string) bool {
+	for _, v := range app_list {
+		if v == app {
+			return true
+		}
+	}
+	return false
+}
+
+func createRuleForBackup(rule_name string, orgID string, app_list []string, pre_post_flag string, ps map[string]map[string]string) bool {
+	pod_selector := []map[string]string{}
+	action_value := []string{}
+	container := []string{}
+	background := []bool{}
+	run_in_single_pod := []bool{}
+	var rulesinfo api.RulesInfo
+	var uid string
+	for i := 0; i < len(app_list); i++ {
+		if pre_post_flag == "pre" {
+			if contains(pre_rule_app, app_list[i]) == true {
+				create_pre_rule = true
+				pod_selector = append(pod_selector, ps[app_list[i]])
+				action_value = append(action_value, app_parameters[app_list[i]]["pre_action_list"])
+				background_val, _ := strconv.ParseBool(app_parameters[app_list[i]]["background"])
+				background = append(background, background_val)
+				pod_val, _ := strconv.ParseBool(app_parameters[app_list[i]]["run_in_single_pod"])
+				run_in_single_pod = append(run_in_single_pod, pod_val)
+				// Here user has to set env for each app container if required in the format container<app name> eg: containersql
+				container_name := fmt.Sprintf("%s-%s", "container", app_list[i])
+				container = append(container, os.Getenv(container_name))
+			}
+		} else {
+			if contains(post_rule_app, app_list[i]) == true {
+				create_post_rule = true
+				pod_selector = append(pod_selector, ps[app_list[i]])
+				action_value = append(action_value, app_parameters[app_list[i]]["post_action_list"])
+				background_val, _ := strconv.ParseBool(app_parameters[app_list[i]]["background"])
+				background = append(background, background_val)
+				pod_val, _ := strconv.ParseBool(app_parameters[app_list[i]]["run_in_single_pod"])
+				run_in_single_pod = append(run_in_single_pod, pod_val)
+				// Here user has to set env for each app container if required in the format container<app name> eg: containersql
+				container_name := fmt.Sprintf("%s-%s", "container", app_list[i])
+				container = append(container, os.Getenv(container_name))
+			}
```

Comment:
@snigdha-px will have to handle the error incase requested app is not found in the list.

Reply:
done